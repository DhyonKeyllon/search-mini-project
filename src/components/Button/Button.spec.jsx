const { render, screen, fireEvent } = require("@testing-library/react");

import { ButtonComponent } from ".";

describe("<ButtonComponent />", () => {
  it('should render the button with text "Load more posts"', () => {
    render(<ButtonComponent text="Load more posts" />);
    expect.assertions(1);

    const button = screen.getByRole("button", { name: /load more posts/i });
    expect(button).toBeInTheDocument();
  });

  it("should call function on click button", () => {
    const fn = jest.fn();
    render(<ButtonComponent text="Load more posts" onClick={fn} />);

    const button = screen.getByRole("button", { name: /load more posts/i });

    fireEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(1);
  });

  it("should be disabled when disabled is true", () => {
    render(<ButtonComponent text="Load more posts" disabled={true} />);

    const button = screen.getByRole("button", { name: /load more posts/i });

    expect(button).toBeDisabled();
  });

  it("should be enabled when disabled is false", () => {
    render(<ButtonComponent text="Load more posts" disabled={false} />);

    const button = screen.getByRole("button", { name: /load more posts/i });

    expect(button).toBeEnabled();
  });

  it("should match snapshot", () => {
    const fn = jest.fn();
    const { container } = render(
      <ButtonComponent text="Load more posts" disabled={false} onClick={fn} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
