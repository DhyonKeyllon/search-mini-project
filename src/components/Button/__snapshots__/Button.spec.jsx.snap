// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonComponent /> should match snapshot 1`] = `
<button
  class="button"
>
  Load more posts
</button>
`;
